CHANGELOG.md

```diff
@@ -3,6 +3,26 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## [0.4.2](https://github.com/paritytech/Nomidot/compare/v0.4.1...v0.4.2) (2020-04-22)
```

Comment:
It's really better to run `learn version *` from master branch. Right now this CHANGELOG shows all your internal commits, which will disappear once we squash this PR, and the v0.4.2 tag won't point to the correct commit.

Reply:
If this workflow is confusing, I'm happy to revisit.

Reply:
ah shoot sorry... I thought that `lerna version` just bumps the version in package.json and Changelog and then we're only to do `lerna publish` from master.

Should we put this in CI?

Reply:
should I revert that commit?

Reply:
> Should we put this in CI?

Yeah, that's what I was thinking: let CI do `yarn lerna version && yarn publish`. Maybe we could put a label `release:{minor,major,patch}`? I'm pretty sure Github Actions can do that easily

Reply:
> should I revert that commit?

That would be the cleanest for future people reading the changelog:
- revert that commit
- delete the 0.4.2 tag from github
- run the whole process again from master once this is merged

Reply:
Alternatively, for the CI, you could what Jaco does: `yarn lerna:version` (without --conventional-commits) manually like in this PR, manually update CHANGELOG, and let CI do the publishing. The scripts for that are already available.

Reply:
> The scripts for that are already available.

I personally prefer your first suggestion, but also what extra scripts are needed for this?

Reply:
If we were to follow exactly Jaco's way, I believe none, they are all here https://github.com/polkadot-js/dev/tree/master/packages/dev/scripts

Reply:
> > should I revert that commit?
> 
> That would be the cleanest for future people reading the changelog:
> 
> * revert that commit
> * delete the 0.4.2 tag from github
> * run the whole process again from master once this is merged

I've so far reverted the commit and deleted the tag from github.

Next I'll get https://github.com/paritytech/Nomidot/pull/314 through and then test by merging this branch

Reply:
if we do add `release:{minor,major,patch}` on PRs to determine the version bump, then we'd no longer use the `--conventional-commits` flag, right (since `lerna` autodetects the bumps based on fix/feat/breaking in commit messages)?

I'm fine with either, but the auto-generated Changelog is quite helpful.

Reply:
https://github.com/paritytech/Nomidot/pull/314

Reply:
You can let lerna autodetect the bump, that's `lerna version --conventional-commits`.

But you can also force lerna to bump as you decide:
- `lerna version patch --conventional-commits` bumps a semver patch (even though some commit messages are feat or have breaking change) and adds an entry in the changelog

So my proposal would be:
- never do `lerna version --conventional-commits`
- always do `lerna version {major,minor,patch} --conventional-commits` based on the PR label `release:{major,minor,patch}`

Reply:
do you think it'd be better to do this via Github Actions, or via Gitlab? I just went with gitlab since that's where all the CI stuff is already.

Reply:
I personally tend to prefer github actions, for 3 reasons: it's faster than gitlab, there's a bigger community (with their marketplace), and I generally can figure things myself (whereas I generally need devops team help with Gitlab).

But both should be able to do the work fine.

Also: if you go with Github, you would need to put your NPM_TOKEN on Github, in their Secrets. I think it's fine (Jaco does it, I did with js-libs before), but it's of course better to store this secret on our own Gitlab.